### DISCUSSION TITLE
Getting 404s with dynamic routes tutorial

### DISCUSSION THREAD

**Discussion by kourtni** [vercel/next.js#22840] [Help]
Hi,

I've been working my way through the Next.js tutorial, and I've gotten everything to work until the [dynamic routes section](https://nextjs.org/learn/basics/dynamic-routes/implement-getstaticprops). For some reason, that I can't figure out, I get 404s when I try to navigate to the routes that should be configured (http://localhost:3000/posts/ssg-ssr and http://localhost:3000/posts/pre-rendering). I've even copied `pages/posts/[id].js` and `lib/posts.js` from the[ Github repo](https://github.com/vercel/next-learn-starter/tree/master/dynamic-routes-step-1), and still get 404s.

In this [post](https://github.com/vercel/next.js/discussions/22590), the author mentions that they were able to resolve this issue by using getInitialProps instead of getStaticProps + getStaticPaths. However, that seems a bit advanced for fixing an issue with the tutorial.

Weirdly, I'm wondering if it's possible that docker isn't playing nicely. The only reason I am considering that at all is because when I first started the tutorial, changes to my files weren't automatically updating in browser and I had to manually restart in order for the changes to show up. Turned out that adding `CHOKIDAR_USEPOLLING=true` to an `.env` file resolved that issue.

Any help here would be appreciated.

Thanks!

**Comment by leerob** [comment]
Hello! Did you try to use the completed example from the tutorial to ensure it wasn't a user error? You could also try running locally without docker, and then with, to verify if it's something with your docker setup. You might have more luck here if you include the source code for a reproducible example. 

There's also Docker docs now: https://nextjs.org/docs/deployment#docker-image
